### ISSUE TITLE
Build instructions for Unix seem out of date

### ISSUE THREAD

**Issue by wcout** [fltk/fltk#67]
README_Unix.txt has not been updated a while. It does not even mention FLTK 1.4 ;)

Especially it would be interesting what to do, after cloning from github.
Typing 'make' works out of the box normally, but what to do, if configuration is needed before?

NOCONFIGURE=1 ./autogen.sh, as stated in README_Unix.txt seems to work (at least a configure script is generated), but the output is somewhat confusing:

configure.ac: error: no proper invocation of AM_INIT_AUTOMAKE was found.
configure.ac: You should verify that configure.ac invokes AM_INIT_AUTOMAKE,
configure.ac: that aclocal.m4 is present in the top-level directory,
configure.ac: and that aclocal.m4 was recently regenerated (using aclocal)
configure.ac:28: installing './config.guess'
configure.ac:28: installing './config.sub'
automake: error: no 'Makefile.am' found for any configure output

Is this still the recommended way?

**Comment by Albrecht-S** [comment]
Thanks for posting this.

> Typing 'make' works out of the box normally, but what to do, if configuration is needed before?

Just to answer this question: you can always use `make clean; ./configure ... (options)`. This adds some overhead but that's what I prefer to do - just because it's simple to remember.

Regarding the instructions: I'll take a look, but I believe that these warnings exist for a long time, even in FLTK 1.3. But I also admit that it would be better not to have these warnings.

That said: my recommendation is to use CMake in out-of-source builds anyway.

**Comment by diplodocuslongus** [comment]
Some updates in the unix readme would be much welcome indeed (or additional clarifications in the present issue).

So far I've gathered 3 different ways to install FLTK (on linux), all tested, FLTK builds fine.

- using the instructions in the README_Unix.txt 
That is 
```
NOCONFIGURE=1 ./autogen.sh
./configure --help
./configure
make
sudo make install
```

I've tested this, builds fine (ubuntu 18.04)

- using the approach mentioned above by @Albrecht-S 
> you can always use make clean; ./configure ... (options)

That is:
After cloning the repo and checked out the proper branch or tag,
`make`
An exectuable configure is created an can be used to configure the build, which implies to first:
`make clean`
Then `./configure [options]`
```
make
sudo make install
```
- using cmake
Create a build directory (from the fltk root)
```
cd build
cmake ..
make
sudo make install
```
No ./configure is available with this approach.

The examples in test/demo have to be build and will be in build/bin/test/examples

It'd be good someone elaborate on the above approaches, clarifies and correct me if needed. 

Worth noting, there's also the approach to download a tar.gz archive from e.g. https://www.fltk.org/software.php?SOFTWARE=v1_1&LOCATION=SourceForge, the extracted archive does contain a ./configure executable which can be used to setup some custom configuration.
Cheers,
Ludo

**Comment by Albrecht-S** [comment]
@diplodocuslongus Hi Ludo, thanks for testing and reporting your results here. The general answer to this issue is that I intend to add another chapter to the docs describing the correct way to build FLTK itself and to build programs with FLTK. I hope I can find the time soon after the release of 1.3.6.

Regarding some of your points:
- using the instructions in the README_Unix.txt

These instructions are really outdated. With the latest version 1.4 (Git master) an easier way is:
```
autoconf
./configure
```
- using the approach mentioned above by @Albrecht-S (me)

The idea here it NOT to run a full `make` in the first step which would build the entire FLTK library and test programs unless you interrupt it with `^C`. Using `make clean` as the first command skips the build step and is much faster, so you would use:
```
make clean
./configure <options> ...
make
```

- using cmake

Instructions are so far correct. There's indeed no `.configure` but there are several CMake options you can set via the commandline or edit in the cache.

- Worth noting, there's also the approach to download a tar.gz archive ... (URL elided)

Where did you get that URL from? The correct URL is: https://www.fltk.org/software.php?VERSION=1.3.5 where `VERSION=...` may be omitted or replaced with the requested version. You still need to select the correct file(s) and trigger the download manually. Note that you can also click the `Download` link on the main FLTK page.

- Not mentioned before: you can also download a Zip archive from GitHub for any version (branch, commit) you like. This archive contains the same files as a Git download, hence you need to run `autoconf` as in points one and two above.

There's much more to say about it (configure and/or CMake options, different CMake generators etc.) which is why I'm planning to add a full chapter to the documentation rather than having lots of README files with unavoidable overlap. For instance `README.CMake.txt` naturally has to cover all available build platforms and `README.Unix.txt` also applies to builds under Windows with Cygwin or MinGW (MSYS).

**Comment by diplodocuslongus** [comment]
@Albrecht-S thanks for the return!

> The idea here it NOT to run a full `make`

Got it! :-)

> Where did you get that URL from?

Most likely from a google search but I can't remember exactly.

PS: the reason why I've tried several different ways to build FLTK was that I couldn't succeed in building mathGL with FLTK backend. The solution was simply to configure FLTK with `--enable-shared` (using the tar.gz archive).

Ludo

**Comment by MatthiasWM** [comment]
All README files must be checked and updated to reflect 1.4 changes and improvements over time. The tutorial on building Editor is slightly out of sync with the app itself. Clarify FLUID handbook situation. Github front page should show sample screenshots and link to apps that use FLTK. Verify all screenshots in the handbook and online, and maybe not use the default Window98 scheme, but Oxy.

**Comment by Albrecht-S** [comment]
@MatthiasWM You added this issue to the 1.4.0 milestone. Although I agree that the README files (and particularly `README.Unix.txt`) need updates I'm too busy with CMake and release stuff. IMHO releasing 1.4.0 is more important than updating the old README files beyond maybe editing version numbers and the most important mistakes or changes since 1.3.x. Everything else can be done whenever we find the time...

Matthias, can you please try to update the most important stuff **or remove this issue from the milestone** again? Thanks.

PS: I just checked `README.Unix.txt` and noticed that a lot of stuff has been updated in Oct. 2022, thus there's probably not much to do now.

**Comment by Albrecht-S** [comment]
Fixed in commit c87e5c755d5a889feb879e89347280b33cfc26ec.

Notes:
- This commit is "as up-to-date as possible" for 1.4.x.
- `README.Unix.txt` will need a major rework in 1.5 anyway when configure will be removed.
